### PR TITLE
feat(retrieval): use healthy peers for requests

### PR DESCRIPTION
**Pull request by istae**
### Checklist

- [ ] I have read the [coding guide](https://github.com/ethersphere/bee/blob/master/CODING.md).
- [ ] My change requires a documentation update, and I have done it.
- [ ] I have added tests to cover my changes.
- [ ] I have filled out the description and linked the related issues.

### Description
Should boost retrieval success rate and performance.

### CODE REVIEW

**Review by notanatol — APPROVED** [review]

File: pkg/retrieval/retrieval.go [hunk 1]
```diff
@@ -364,7 +364,7 @@ func (s *Service) prepareCredit(ctx context.Context, peer, chunk swarm.Address,
 // retrieve request.
 func (s *Service) closestPeer(addr swarm.Address, skipPeers []swarm.Address, allowUpstream bool) (swarm.Address, error) {
 
-	closest, err := s.peerSuggester.ClosestPeer(addr, false, topology.Select{Reachable: true}, skipPeers...)
+	closest, err := s.peerSuggester.ClosestPeer(addr, false, topology.Select{Reachable: true, Healthy: true}, skipPeers...)
```
Comment (aloknerurkar) [inline pkg/retrieval/retrieval.go:367]: Is this required? Are we seeing failed retrieval attempts?

I am just afraid of restricting to only healthy peers here. We could have reachable peers who could potentially respond but wont be considered.

Comment (istae) [reply pkg/retrieval/retrieval.go:367]: On occasion, yes, there are failed downloads, but still a very small rate, like two or three failed downloads per day, out of like 500 smoke test downloads. the other issue is download speed is lagging behind uploads, which is counterintuitive. 

adding healthy peer checks to pushsync improved retrieval as well but adding this check to retrieval is the next step.

using only healthy peers would boost performance and reliability.

Comment (istae) [reply pkg/retrieval/retrieval.go:367]: we can put this on hold until after the localstore release if you'd like

**Review by aloknerurkar — APPROVED** [review]
I am inclined to try this out as retrievals are slow and pushsync is really fast!

**Review by mrekucci — APPROVED** [review]

**Pull request merged**